src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatibilityAnalyzerTests.GuardedCallsTests.cs

```diff
@@ -3945,6 +3945,454 @@ bool IsWindows11OrLater()
             await VerifyAnalyzerAsyncCs(source, "dotnet_code_quality.interprocedural_analysis_kind = ContextSensitive");
         }
 
+        [Fact]
+        public async Task GuardMembersWithSupportedGuardAttributes()
+        {
+            var source = @"
+using System;
+using System.Diagnostics;
+using System.Runtime.Versioning;
+
+class Test
+{
+    [SupportedOSPlatformGuard(""linux"")]
+    private bool IsLunuxSupported() => true;
+
+    [SupportedOSPlatformGuard(""linux"")]
+    [SupportedOSPlatformGuard(""Windows"")]
+    internal bool LinuxAndWindowsSupported { get; set; }
+
+    [SupportedOSPlatformGuard(""linux"")]
+    [SupportedOSPlatformGuard(""macOS"")]
+    [SupportedOSPlatformGuard(""Windows"")]
+    private readonly bool _http3Enabled;
+
+    [SupportedOSPlatformGuard(""linux"")]
+    [SupportedOSPlatformGuard(""macOS"")]
+    [SupportedOSPlatformGuard(""Windows"")]
+    [SupportedOSPlatformGuard(""Android"")]
+    private bool IsHttp3PlusAndroidEnabled() => false;
+
+    [SupportedOSPlatformGuard(""linux"")]
+    [SupportedOSPlatformGuard(""macOS"")]
+    [SupportedOSPlatformGuard(""Windows"")]
+    [UnsupportedOSPlatformGuard(""Android"")]
+    private readonly bool _http3EnabledNotAndroid;
+
+    void M1()
+    {
+        if (IsLunuxSupported()) // one of the support guarded, so no warning
```

Comment:
Just trying to understand the desired semantics here - so we are not actually performing flow analysis on the called method, `IsLunuxSupported` here, but just decoding the new platform guard attributes on the method to drive the analysis, correct? Basically, if I were to change this test case so `IsLunuxSupported` returned `false` instead of `true` and made no other changes, will the analysis + diagnostics stay exactly the same?

Reply:
> Just trying to understand the desired semantics here - so we are not actually performing flow analysis on the called method, IsLunuxSupported here, but just decoding the new platform guard attributes on the method to drive the analysis, correct

Correct

> Basically, if I were to change this test case so IsLunuxSupported returned false instead of true and made no other changes, will the analysis + diagnostics stay exactly the same?

Right, no matter what value the method returns explicitly, only matter what attribute it has with what platform and or version, then it can guard corresponding platform-specific APIs on its `true` side
```cs
[SupportedOSPlatformGuard("linux")]
private bool IsLunuxSupported() => false; // same guard
```

Reply:
Sounds good, thanks!